src/main/java/net/neoforged/neoforge/client/settings/KeyModifier.java

```diff
@@ -129,4 +183,9 @@ public static KeyModifier valueFromString(String stringValue) {
     public abstract boolean isActive(@Nullable IKeyConflictContext conflictContext);
 
     public abstract Component getCombinedName(InputConstants.Key key, Supplier<Component> defaultLogic);
+
+    // Neo: Make abstract in 1.21.5
+    public InputConstants.Key[] codes() {
+        return null;
```

Comment:
Seeing a `null` in a non-nullable place gives me the 👀, but since there's no genuine way this `null` can be received (without some enum-extending shenaniganry), this should be fine.

Reply:
The reason I did this was to avoid breaking changes, but yeah, it should generally be changed in 1.21.5 to avoid such a null case.